control-plane/connect-inject/controllers/pod/pod_controller_test.go

```diff
@@ -1281,19 +1301,24 @@ func TestReconcileDeletePod(t *testing.T) {
 	}
 }
 
-// createPod creates a multi-port pod as a base for tests.
-func createPod(name, ip string, identity string, inject bool, ready bool) *corev1.Pod {
+// createPod creates a multi-port pod as a base for tests. If `namespace` is empty,
+// the default Kube namespace will be used.
+func createPod(name, namespace string, inject, ready bool) *corev1.Pod {
+	if namespace == "" {
+		namespace = metav1.NamespaceDefault
+	}
+
 	pod := &corev1.Pod{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      name,
-			Namespace: metav1.NamespaceDefault,
+			Namespace: namespace,
 			Labels:    map[string]string{},
 			Annotations: map[string]string{
 				constants.AnnotationConsulK8sVersion: "1.3.0",
 			},
 		},
 		Status: corev1.PodStatus{
-			PodIP:  ip,
+			PodIP:  "10.0.0.1",
```

Comment:
~~ I think every pod technically gets its own IP address. Seems this is always called once and won't matter for these tests, but could comment that it's not meant for multiple-pod use?